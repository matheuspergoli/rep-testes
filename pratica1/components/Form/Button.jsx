import React from 'react'

function Button() {
	return (
		<button
			type='submit'
			className='p-2 uppercase rounded-md font-bold text-white bg-btnGreen'>
			Claim your free trial
		</button>
	)
}

export default Button
