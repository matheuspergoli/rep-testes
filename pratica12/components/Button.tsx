import React from 'react'

interface ButtonProps {
	className: string
}

function Button(props: ButtonProps) {
	return (
		<button
			className={`font-bold my-10 py-2 px-6 rounded-full bg-white ${props.className}`}>
			Acessar
		</button>
	)
}

export default Button
