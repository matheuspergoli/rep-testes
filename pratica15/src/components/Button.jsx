import React from 'react'

function Button() {
	return (
		<button className='px-4 py-1 rounded-md transition font-semibold bg-gray-400 hover:bg-gray-700 hover:text-white active:scale-95'>
			Clique aqui
		</button>
	)
}

export default Button
