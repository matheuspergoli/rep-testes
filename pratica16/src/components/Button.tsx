import React from 'react'

interface ButtonProps {
	text: string
}

function Button(props: ButtonProps) {
	return (
		<button className='mt-5 px-5 py-1 font-semibold transition rounded-full bg-gray-600 hover:bg-gray-700 active:scale-95'>
			{props.text}
		</button>
	)
}

export default Button
