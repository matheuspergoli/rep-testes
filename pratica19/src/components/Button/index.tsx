import React from 'react'

export interface ButtonProps {
	font: string
	label: string
	radius: string
	background: string
}

function Button(props: ButtonProps) {
	return (
		<button
			className={`px-4 py-1 text-white ${props.background} ${props.radius} ${props.font}`}
		>
			{props.label}
		</button>
	)
}

export default Button
