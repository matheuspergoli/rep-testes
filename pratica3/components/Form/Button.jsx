import React from 'react'

function Button() {
	return (
		<button className='rounded-md mb-10 p-2 bg-btn-color' type='submit'>
			Acessar
		</button>
	)
}

export default Button
