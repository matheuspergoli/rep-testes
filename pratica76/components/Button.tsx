import React from 'react'

enum ClassNames {
	primary = 'bg-blue-500 hover:bg-blue-700 text-white py-2 px-4 rounded transition disabled:bg-opacity-50 disabled:cursor-not-allowed',
	secondary = 'bg-gray-500 hover:bg-gray-700 text-white py-2 px-4 rounded transition disabled:bg-opacity-50 disabled:cursor-not-allowed'
}

interface ButtonProps {
	/** Button label */
	label: string

	/** Button disabled state */
	disabled?: boolean

	/** Button loading state */
	loading?: boolean

	/** Button type */
	type?: 'button' | 'submit' | 'reset'

	/** Button value */
	value?: string

	/** Button className */
	variant?: 'primary' | 'secondary'
}

export const Button = (props: ButtonProps) => {
	return (
		<button
			className={ClassNames[props.variant || 'primary']}
			disabled={props.disabled}
			type={props.type}
			value={props.value}>
			{props.loading ? 'Loading...' : props.label}
		</button>
	)
}
