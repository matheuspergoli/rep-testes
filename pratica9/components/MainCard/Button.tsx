import React from 'react'

interface ButtonProps {
	text: string
	onClick: () => void
}

function Button(props: ButtonProps) {
	return (
		<button
			className='w-full p-2 border rounded-md bg-black text-white'
			type='submit'
			onClick={props.onClick}>
			{props.text}
		</button>
	)
}

export default Button
