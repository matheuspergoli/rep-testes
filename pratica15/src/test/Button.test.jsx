import { describe, expect, it, vitest } from 'vitest'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Button from '../components/Button'

describe('<Button />', () => {
	it('Should render', () => {
		render(<Button />)
		const button = screen.getByRole('button', { name: 'Clique aqui!' })
		expect.assertions(1)
		expect(button).toBeInTheDocument()
	})

	it('Should call function on click', async () => {
		const fn = vitest.fn()
		render(<Button onClick={fn} />)
		const button = screen.getByRole('button', { name: 'Clique aqui!' })
		expect.assertions(1)
		await userEvent.click(button)
		await userEvent.click(button)
		expect(fn).toBeCalledTimes(2)
	})

	it('Should have the class active:scale-95', () => {
		render(<Button />)
		const button = screen.getByRole('button', { name: 'Clique aqui!' })
		expect.assertions(1)
		expect(button).toHaveClass('active:scale-95')
	})

	it('Should match the snapshot', () => {
		render(<Button />)
		const button = screen.getByRole('button', { name: 'Clique aqui!' })
		expect.assertions(1)
		expect(button).toMatchSnapshot()
	})
})
