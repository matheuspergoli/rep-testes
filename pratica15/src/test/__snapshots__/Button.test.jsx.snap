// Vitest Snapshot v1

exports[`<Button /> > Should match the snapshot 1`] = `
<button
  class="px-4 py-1 rounded-md transition font-semibold bg-gray-400 hover:bg-gray-700 hover:text-white active:scale-95"
>
  Clique aqui!
</button>
`;
